eve/main.yml

```diff
@@ -143,3 +144,70 @@ stages:
             echo "%(secret:ssh_pub_keys)s" >> ~/.ssh/authorized_keys &&
             sleep 3600
           doStepIf: false
+
+  multiple-nodes:
+    worker:
+      type: openstack
+      image: CentOS 7 (PVHVM)
+      flavor: m1.medium
+      path: eve/workers/openstack-multiple-nodes
+    steps:
+      - Git: *git_pull
+      - ShellCommand:
+          name: Init terraform
+          command: "terraform init"
+          workdir: "build/eve/workers/openstack-multiple-nodes/terraform/"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Validate terraform definition
+          command: "terraform validate"
+          workdir: "build/eve/workers/openstack-multiple-nodes/terraform/"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Spawn openstack virtual infra
+          command: 'terraform apply -auto-approve'
+          workdir: "build/eve/workers/openstack-multiple-nodes/terraform/"
+          env: &terraform_spawn_multiple_nodes
+            OS_AUTH_URL: "%(secret:scality_cloud_auth_url)s"
+            OS_REGION_NAME: "%(secret:scality_cloud_region)s"
+            OS_USERNAME: "%(secret:scality_cloud_username)s"
+            OS_PASSWORD: "%(secret:scality_cloud_password)s"
+            OS_TENANT_NAME: "%(secret:scality_cloud_tenant_name)s"
+            TF_VAR_worker_uuid: "%(prop:worker_uuid)s"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Configure SSH config for bootstrap node
+          command: >
+            sed -i s/IP_ADDRESS/$(terraform output bootstrap_ip)/
+            bootstrap_ssh_config
+          workdir: "build/eve/workers/openstack-multiple-nodes/terraform/"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check SSH config for bootstrap node
+          command: |-
+            for _ in $(seq 1 12); do
+              sleep 5
+              ssh -F bootstrap_ssh_config bootstrap id && break
+            done;
+          workdir: "build/eve/workers/openstack-multiple-nodes/terraform/"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Destroy openstack virtual infra
+          command: |-
+            for _ in $(seq 1 3); do
+               terraform destroy -auto-approve && break
+            done;
+          workdir: "build/eve/workers/openstack-multiple-nodes/terraform/"
+          env: *terraform_spawn_multiple_nodes
+          alwaysRun: true
+          sigtermTime: 600
+      - ShellCommand:
```

Comment:
Depends on what you want to debug. And since any debugging will need a manual edition of the file, then I think moving the debug step wherever you want is acceptable.
I'm perfectly OK if in a next PR we move the debug step before the destruction though.